sdk/storage/Azure.Storage.Blobs/tests/BlobTestBase.cs

```diff
@@ -246,75 +246,109 @@ public StorageSharedKeyCredential GetNewSharedKeyCredentials()
+        }
 
         public BlobSasQueryParameters GetNewBlobServiceSasCredentialsContainer(string containerName, StorageSharedKeyCredential sharedKeyCredentials = default)
-            => new BlobSasBuilder
+        {
+            var builder = new BlobSasBuilder
             {
                 BlobContainerName = containerName,
                 Protocol = SasProtocol.None,
                 StartsOn = Recording.UtcNow.AddHours(-1),
                 ExpiresOn = Recording.UtcNow.AddHours(+1),
-                Permissions = new BlobContainerSasPermissions { Read = true, Add = true, Create = true, Write = true, Delete = true, List = true }.ToString(),
                 IPRange = new SasIPRange(IPAddress.None, IPAddress.None)
-            }.ToSasQueryParameters(sharedKeyCredentials ?? GetNewSharedKeyCredentials());
+            };
+            builder.SetPermissions(BlobContainerSasPermissions.All);
+            return builder.ToSasQueryParameters(sharedKeyCredentials ?? GetNewSharedKeyCredentials());
+        }
 
         public BlobSasQueryParameters GetNewBlobServiceIdentitySasCredentialsContainer(string containerName, UserDelegationKey userDelegationKey, string accountName)
-            => new BlobSasBuilder
+        {
+            var builder = new BlobSasBuilder
             {
                 BlobContainerName = containerName,
                 Protocol = SasProtocol.None,
                 StartsOn = Recording.UtcNow.AddHours(-1),
                 ExpiresOn = Recording.UtcNow.AddHours(+1),
-                Permissions = new BlobContainerSasPermissions { Read = true, Add = true, Create = true, Write = true, Delete = true, List = true }.ToString(),
                 IPRange = new SasIPRange(IPAddress.None, IPAddress.None)
-            }.ToSasQueryParameters(userDelegationKey, accountName);
+            };
+            builder.SetPermissions(BlobContainerSasPermissions.All);
+            return builder.ToSasQueryParameters(userDelegationKey, accountName);
+        }
 
         public BlobSasQueryParameters GetNewBlobServiceSasCredentialsBlob(string containerName, string blobName, StorageSharedKeyCredential sharedKeyCredentials = default)
-            => new BlobSasBuilder
+        {
+            var builder = new BlobSasBuilder
             {
                 BlobContainerName = containerName,
                 BlobName = blobName,
                 Protocol = SasProtocol.None,
                 StartsOn = Recording.UtcNow.AddHours(-1),
                 ExpiresOn = Recording.UtcNow.AddHours(+1),
-                Permissions = new BlobSasPermissions { Read = true, Add = true, Create = true, Write = true, Delete = true }.ToString(),
                 IPRange = new SasIPRange(IPAddress.None, IPAddress.None)
-            }.ToSasQueryParameters(sharedKeyCredentials ?? GetNewSharedKeyCredentials());
+            };
+            builder.SetPermissions(
+                BlobContainerSasPermissions.Read |
```

Comment:
This was `BlobSasPermissions` but you're using `BlobContainerSasPermissions` here?

Reply:
Fixed